app/src/main/java/com/github/se/eventradar/viewmodel/SearchFilterUiState.kt

```diff
@@ -0,0 +1,15 @@
+package com.github.se.eventradar.viewmodel
+
+import com.github.se.eventradar.model.event.EventCategory
+
+// This class represents a common uiState for search and filter operations
+// It allows uiState in FilterPopUp to be of either type EventsOverviewUiState or HostedEventUiState
+sealed class SearchFilterUiState {
```

Comment:
works for me! this keeps the state in the search bar and filter upon switching tabs in the app, which i think may not be the same if it's a normal data class 👍